subworkflows/canine_lancet_module.cwl

```diff
@@ -0,0 +1,141 @@
+
+inputs:
+  # Killswitch
+  disable_workflow: { type: 'boolean?', doc: "For when this workflow is wrapped into a larger workflow, you can use this value in the when statement to toggle the running of this workflow." }
+
+  calling_intervals: { type: 'File', doc: "YAML file contianing the intervals in which to perform variant calling." }
+  indexed_reference_fasta: { type: 'File', secondaryFiles: [{ pattern: '.fai', required: true }], doc: "Reference fasta and fai index" }
+  input_tumor_reads: { type: 'File', secondaryFiles: [{ pattern: '.bai', required: false}, { pattern: '^.bai', required: false}], doc: "BAM file containing mapped reads from the tumor sample" }
+  input_normal_reads: { type: 'File', secondaryFiles: [{ pattern: '.bai', required: false}, { pattern: '^.bai', required: false}], doc: "BAM file containing mapped reads from the normal sample" }
+  output_basename: { type: 'string', doc: "String to use as basename for outputs." }
+  targets_file: { type: 'File?', doc: "For exome variant calling, this file contains the targets regions used in library preparation." }
+
+  # Resource Control
+  lancet_ram: { type: 'int?', doc: "Maximum GB of RAM to allocate to Lancet." }
+  lancet_cpu: { type: 'int?', doc: "Number of CPUs to allocate to Lancet." }
+
+outputs:
+  lancet_all_vcf: { type: 'File', outputSource: bcftools_concat_sort_index/output }
+  lancet_pass_vcf: { type: 'File', outputSource: bcftools_filter_index/output }
+  lancet_all_vcf_stats: { type: 'File', outputSource: bcftools_stats_all/stats }
+  lancet_pass_vcf_stats: { type: 'File', outputSource: bcftools_stats_pass/stats }
+
+steps:
+  expr_conditional:
+    run: ../tools/expr_conditional.cwl
+    when: $(inputs.disable == true)
+    in:
+      disable: disable_workflow
+    out: [output]
+
+  calling_intervals_yaml_to_beds:
+    run: ../tools/calling_intervals_yaml_to_beds.cwl
+    in:
+      input_yaml: calling_intervals
+    out: [outputs]
+
+  lancet:
+    hints:
+      - class: 'sbg:AWSInstanceType'
+        value: c5.9xlarge
+    run: ../tools/lancet.cwl
+    scatter: [bed]
+    in:
+      normal: input_normal_reads
+      tumor: input_tumor_reads
+      ref: indexed_reference_fasta
+      bed: calling_intervals_yaml_to_beds/outputs
+      output_filename:
+        source: output_basename
+        valueFrom: $(self).$(inputs.bed.nameroot).lancet-uns.vcf
+      max_vaf_normal:
+        valueFrom: $(0.05)
+      max_alt_count_normal:
+        valueFrom: $(50)
+      cpu: lancet_cpu
+      ram: lancet_ram
+    out: [vcf]
+
+  bcftools_reheader_sort_index:
+    run: ../tools/bcftools_reheader_sort_index.cwl
+    scatter: [input_file]
+    in:
+      input_file: lancet/vcf 
+      output_filename:
+        valueFrom: $(inputs.input_file.nameroot).sorted.vcf
+      output_type:
+        valueFrom: "v"
+      fai:
+        source: indexed_reference_fasta
+        valueFrom: |
+          $(self.secondaryFiles.filter(function(e){ return e.nameext == '.fai' })[0])
```

Comment:
nifty!

Reply:
Yeah was surprised with this one worked without dying.